code-reviews/recipes/Bash.md

```diff
@@ -22,6 +64,57 @@ To automate this process in Azure Devops you can add the following snippet to yo
     shellcheck ./scripts/*.sh
   displayName: "Validate Scripts: Shellcheck"
   ```
+  
+  Also, your shell scripts can be formatted in your build pipeline by using shfmt. To integrate shfmt in your build pipeline do the following:
+
+```yaml
+- bash: |
+    echo "This step does auto formatting of shell scripts"
+    shfmt -l -w ./scripts/*.sh
+  displayName: "Format Scripts: shfmt"
+  ```
+  
+  Unit testing using [shunit2](https://github.com/kward/shunit2) can also be added to the build pipeline, using the following block:
+  
+  ```yaml
+- bash: |
+    echo "This step unit tests shell scripts by using shunit2"
+    ./shunit2
+  displayName: "Format Scripts: shfmt"
+  ```
+  
+## Pre-Commit Hooks
+
+All developers should run shellcheck and shfmt as pre-commit hooks.
+
+### Step 1- Install pre-commit
+
+Run `pip install pre-commit` to install pre-commit.
+Alternatively you can run `brew install pre-commit` if you are using homebrew.
+
+### Step 2- Add shellcheck and shfmt
+
+Add .pre-commit-config.yaml file to root of the go project. Run shfmt on pre-commit by adding it to .pre-commit-config.yaml file like below.
+
+```yaml
+-   repo: git://github.com/pecigonzalo/pre-commit-fmt
+    sha: master
+    hooks:
+      -   id: shell-fmt
+          args:
+            - --indent=4
+ ```
+
+ ```yaml
+-   repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.7.1.1
+    hooks:
+    -   id: shellcheck
+ ```
+
+### Step 3
+
+Run `$ pre-commit install` to setup the git hook scripts
 
 ## Dependencies
```

Comment:
The Code Review Checklist is still empty. Something we can add here?

Reply:
Done.